rewrite-maven/src/main/java/org/openrewrite/maven/AddGradleEnterpriseMavenExtension.java

```diff
@@ -158,6 +183,59 @@ protected List<SourceFile> visit(List<SourceFile> before, ExecutionContext ctx)
         return ListUtils.concat(ListUtils.concat(before, extensionsXml), gradleEnterpriseXml);
     }
 
+    @JacksonXmlRootElement(localName = "gradleEnterprise")
+    @Value
+    private static class GradleEnterpriseConfiguration {
+        ServerConfiguration server;
+        @Nullable
+        BuildScanConfiguration buildScan;
+    }
+
+    @Value
+    private static class ServerConfiguration {
+        String url;
+        @Nullable
+        Boolean allowUntrusted;
+    }
+
+    @Value
+    private static class BuildScanConfiguration {
+        @Nullable
+        Boolean backgroundBuildScanUpload;
+        @Nullable
+        String publish;
+        @Nullable
+        Capture capture;
+    }
+
+    @Value
+    private static class Capture {
+        Boolean goalInputFiles;
+    }
+
+    private String gradleEnterpriseConfiguration() {
+        BuildScanConfiguration buildScanConfiguration = buildScanConfiguration();
+        ServerConfiguration serverConfiguration = new ServerConfiguration(server, allowUntrustedServer);
+        try {
+            ObjectMapper objectMapper = MavenXmlMapper.writeMapper();
+            objectMapper.enable(SerializationFeature.INDENT_OUTPUT);
+            objectMapper.setSerializationInclusion(JsonInclude.Include.NON_ABSENT);
+            return objectMapper.writeValueAsString(new GradleEnterpriseConfiguration(serverConfiguration, buildScanConfiguration));
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
```

Comment:
Not sure how those kind of exceptions are usually handled in recipes ?

Reply:
I'm seeing a mix of [`ctx.getOnError().accept(e)`](https://github.com/search?q=org%3Aopenrewrite%20ctx.getOnError&type=code), [`Markup.error(e)`](https://github.com/search?q=org%3Aopenrewrite+Markup.error&type=code) and [`throw new RuntimeException`](https://github.com/search?q=org%3Aopenrewrite+%22throw+new+RuntimeException%22&type=code), and possibly others.

I think it will mostly depend on if your recipe will be executed as part of a larger chain, in which case you might want to go for "elegant" handing over just throwing a runtime exception. I'm guessing in this case you're fine with a RuntimeException, since we're also not expecting a lot of different failure modes here.